### PR TITLE
feat: add support for associating tags with a feature flag

### PR DESCRIPTION
**Pull request by weyert**
## Changes

Updates the Add Feature Flag page to allow associating one or more tags
with a feature flag

## How did you test this code?

I have manually executed the following steps:

  - Add tag, and save feature flag
  - Reload feature flag
  - Remove existing tag and save feature flag

### CODE REVIEW

**Comment by weyert** [comment]
Screenshot:

CleanShot 2021-12-01 at 23 39 19@2x

**Comment by weyert** [comment]
@paolodamico @macobo @timgl I have made a PR to introduce basic support for associating tags with feature flag. Let me know, what you think of it.

My main open question how would filtering by tags work for on the feature flags overview page?

**Comment by neilkakkar** [comment]
Hey Weyert! Thanks for PR-ing! I'm a bit out of the loop: What's the use-case for tags here? Did you face a problem that's being solved by these?

**Comment by weyert** [comment]
Mainly organising of the feature flags; so can filter them by tag, e.g. like `ops`, `development`, `experiment` tags :)

Solves my future problem, which `ops` flag do we have? How many `ops` flags are currently enabled? Which feature flags are associated with `backend` or `project1` etc.

**Comment by paolodamico** [comment]
Thanks for putting this out @weyert, I think it'll make it easier to keep things organized. For consistency, filtering should look the same way we have it for saved insights (hope this answers your question). Before continuing with this, I do need to check internally how this fits with all the changes we're making to experimentation, hope that's okay.

**Comment by paolodamico** [comment]
Also just friendly reminder @weyert to follow the same pattern as we have on Dashboards for tags regarding the feature structure and requiring a paid license to use.

**Comment by weyert** [comment]
Added the ability to filter the feature flags by a selected tag and  put the functionality behind a feature flag named `feature-flag-tags` :)

image
BC1/cleanshot_2021-12-06_at_17.14.19.gif

I haven't been able to reproduce the filter bar from your screenshot yet but I will revisit that later. If anyone have any pointers were to look.

File: frontend/src/scenes/feature-flags/featureFlagsLogic.ts [hunk 1]
```diff
@@ -72,6 +96,7 @@ export const featureFlagsLogic = kea<featureFlagsLogicType>({
     events: ({ actions }) => ({
         afterMount: () => {
             actions.loadFeatureFlags()
+            actions.loadFeatureFlagTags()
```
Comment (neilkakkar) [inline frontend/src/scenes/feature-flags/featureFlagsLogic.ts:99]: this needs the FeatureFlag enabled check as well

File: frontend/src/scenes/feature-flags/featureFlagLogic.tsx [hunk 2]
```diff
@@ -290,6 +293,26 @@ export const featureFlagLogic = kea<featureFlagLogicType>({
                 actions.setMultivariateOptions(null)
             }
         },
+        saveNewTag: ({ tag }) => {
```
Comment (neilkakkar) [inline frontend/src/scenes/feature-flags/featureFlagLogic.tsx:296]: think `featureFlagTags` should listen on this as well? Add this tag to the list when created

**Comment by weyert** [comment]
@posthog-bot  still working on it; just on vacation :)

**Pull request closed without merging**